### ISSUE TITLE
Review

### ISSUE THREAD

**Issue by tcrasset** [JonathanDecleire/DL-wine-quality#1]
# Structured data review

You did a fantastic job, that's why I allowed myself to be a bit more finnicky with the things to improve. Hope you see it as a way to be EVEN better

## Good

- Love the `packagelist.txt`
- Nice baseline
- Nice EDA (could have been in a separate file)
- Really like the `fit` and `evaluate` method
- I like that you used a validation set and you plot the loss of both
- I like the confusion matrix
- I like that you check the distribution of each class in the different train/val/test sets
- I really like the overview in the README !

## To improve

- You say you get an accuracy of `60%`, but what you print is the loss (in `lr_pytorch_mode_60` ):

```python
'''
Using linear regression with the pytorch module we are able to improve our score
from 30% with linear regression with the sklearn module to 60%.
It is still not great. Perhaps the data is not linearly related enough to make good
predictions.
'''

result = evaluate(model, val_loader)
print(result)
```

leads to --> `{'val_loss': 0.6495099067687988}`

- You could have done the data preprocessing in a separate function/file so that you diminish code duplication across your notebooks
- You never test on the test set. You should have 3 data sets : train, validation, test. Validation is used for hyper-parameter optimization. Testing set is used at the very end. --> actually you did in `multiclass_pytorch_model_40.ipynb`

- There surely is a better way to Count the class direction that `get_class_distribution`

- `"From there we calculated that by random chance we had a 100/7 = 14.3% chance of guessing the correct score."`
  --> That is only correct if the classes are balanced, which they are not. But ok, that's a first approximation.

- `"As the purpose of the exercise is to use the deep learning module of pytorch in our case, we moved on to trying it out by using a linear regression model again"`
  - What do you mean by linear regression ? If you are using a neural network, by definition it is because you want to model **non-linear** relationships (activation functions such as sigmoid, ReLU are _non-linear_)
